.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,133 @@
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+--.*"
+      - "v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+--.*"
+      - "v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+--.*"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+--.*"
+  # trigger on pull request updates when target is `main` branch
+  pull_request:
+    branches:
+      - "main"
+env:      
+  REGISTRY: ghcr.io
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      conductor: ${{ steps.filters.outputs.conductor }}
+      relayer: ${{ steps.filters.outputs.relayer }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filters
+        with:
+          filters: |
+            conductor:
+              - 'crates/astria-conductor/**'
+              - '.github/workflows/docker-build.yml'
+            relayer:
+              - 'crates/astria-sequencer-relayer/**'
+              - '.github/workflows/docker-build.yml'
+
+  conductor-build-and-publish-latest:
+    runs-on: ubuntu-latest
+    needs: changes
+    if: ${{ needs.changes.outputs.conductor == 'true' || contains(github.ref, 'refs/tags/v') && contains(github.ref, '--conductor') }}
+    steps:
+      # Checking out the repo
+      - uses: actions/checkout@v3
+      # https://github.com/docker/setup-qemu-action
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Log in to registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # Generate correct tabs and labels
+      - name: Docker metadata
+        id: metadata
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ghcr.io/astriaorg/astria-conductor
+          tags: |
+            type=ref,event=pr
+            type=semver,pattern={{major}}.{{minor}}.{{patch}}
+            type=sha
+            # set latest tag for `main` branch
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+      - name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          # this gets rid of the unknown/unknown image that is created without this setting
+          # https://github.com/docker/build-push-action/issues/820#issuecomment-1455687416
+          provenance: false
+          context: .
+          file: crates/astria-conductor/Dockerfile
+          # It takes over 30 minutes to build the arm image right now, so we only build it on tags which is what we use for releases.
+          platforms: ${{ contains(github.ref, 'refs/tags/v') && contains(github.ref, '--conductor') && 'linux/amd64,linux/arm64' || 'linux/amd64' }}
```

Comment:
I've updated this, arm64 still takes ~50% more time. I'm leaning towards "only do on main" for that? This would be more frequent than only on tags, but shorten PR time.